### ISSUE TITLE
Router.Use Not Working with OPTIONS

### ISSUE THREAD

**Issue by nahtnam** [alexmingoia/koa-router#482]
node.js version: v8.11.3

npm/yarn and version: NPM 6.2.0

`koa-router` version: 7.4.0

`koa` version: 2.5.2

#### Code sample:

Requires the three packages imported at the top.
```js
const Koa = require('koa');
const Router = require('koa-router');
const cors = require('@koa/cors');

const app = new Koa();
const router = new Router();

router.use('/test', async (ctx, next) => {
  console.log('this isnt printing for OPTIONS requests! :(');
  await next();
})
router.use('/test', cors()); // doesnt work for OPTIONS either, big problem for me personally

router.post('/test', (ctx) => {
  ctx.body = { hello: 'world' };
});

app.use(router.routes());
app.use(router.allowedMethods());

app.listen(3000)
```

#### Expected Behavior:

When you make an OPTIONS request, the two middleware should be run.
#### Actual Behavior:

The two middleware defined only run on POST. They do not run on OPTIONS.
#### Additional steps, HTTP request details, or  to reproduce the behavior or a test case:

N/A.

**Comment by hcz** [comment]
It is not related to `koa-router`, the list of allowed methods for `@koa/cors` is defined [here](https://github.com/koajs/cors/blob/master/index.js#L19) and you can modify it
````javascript
const Koa = require('koa');
const Router = require('koa-router');
const cors = require('@koa/cors');

const corsOptions = {
  allowMethods: 'POST,OPTIONS'
}

const app = new Koa();
const router = new Router();

router.all('/test', async (ctx, next) => {
  console.log('Options request');
  await next();
});

router.use(cors(corsOptions));

router.post('/test', (ctx) => {
  ctx.body = { hello: 'world' };
});

app.use(router.routes());
app.use(router.allowedMethods());

app.listen(3000)
````

**Comment by mk0y** [comment]
Can this be used per route, not for all routes?

**Comment by hcz** [comment]
> Can this be used per route, not for all routes?

@mk0y This is a case for the nested routers, isn't it?

**Comment by mk0y** [comment]
Not necessarily, but I see now that `someNestedRoute.use(cosrs())` can be used.
On the other hand for example:
```js
forums.use('/forums/:fid/posts', posts.routes(), posts.allowedMethods())
forums.use('/forums/somethingelse', posts.routes(), posts.allowedMethods()) // additionally cors?
```

So don't see a way to use per route, maybe I'm being too specific. I guess `somethingelse` route can be split into nested route.

**Comment by heartwhite** [comment]
hi. 
`app.use(router.allowedMethods());`
does this code allows all methods? 
I am getting an interesting response from my server. 
`response {type: "cors", status:200}`
my server receives request but when it sends response, on frontend I can't see my response. Posman can get the response but on browser I get an response with nothing in it. do you have any idea? I paste the  response on browser I got. 
`Response {type: "cors", url: "https://b169f88dfed4.ngrok.io/api/products?shop=my-embedded-app-test.myshopify.com", redirected: false, status: 200, ok: true, …}
body: (...)
bodyUsed: true
headers: Headers {}
ok: true
redirected: false
status: 200
statusText: "OK"
type: "cors"
url: "https://b169f88dfed4.ngrok.io/api/products?shop=my-embedded-app-test.myshopify.com"
__proto__: Response`
